app/src/main/java/org/covidwatch/android/data/signedreport/firestore/SignedReportsUploader.kt

```diff
@@ -0,0 +1,129 @@
+package org.covidwatch.android.data.signedreport.firestore
+
+import android.util.Base64
+import android.util.Log
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+import okhttp3.MediaType
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import org.covidwatch.android.BuildConfig
+import org.covidwatch.android.data.signedreport.SignedReport
+import org.covidwatch.android.data.signedreport.SignedReportDAO
+import org.json.JSONObject
+import java.io.IOException
+
+class SignedReportsUploader(
+    private val okHttpClient: OkHttpClient,
+    private val signedReportDAO: SignedReportDAO
+) {
+
+    fun startUploading() {
+        GlobalScope.launch(Dispatchers.IO) {
+            signedReportDAO.all().collect {
+                uploadSignedReportsNeeded(it)
```

Comment:
`IfNeeded`?

Reply:
I general, I'm not fond of this Uploader, Worker, etc stuff. We would need to refactor this data layer to the Repository pattern.  I'm saying this because eventually all this code might be gone in the near future so renaming methods is not really matters.

That's being said, I'll rename it because I need to do some subtle changes in the code anyway  :)